### DISCUSSION TITLE
Database / Broadcast notifications: accept Collection and array as parameters

### DISCUSSION THREAD

**Discussion by kamilst96** [filamentphp/filament#3964] [Ideas]
Sometimes we'd fetch users based on a query and push a notification to all of them. Since it involves iterating over it every time, would it be acceptable to implement it in `Notification` class?

Dirty example below; note lack of type safeguards when passing Collection. Should it type check each model and throw if invalid?

```php
namespace Filament\Notifications;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Contracts\Auth\Authenticatable;
use Illuminate\Support\Collection;

class Notification extends ViewComponent implements Arrayable
{
    public function sendToDatabase(Model | Authenticatable | Collection | array $users): static
    {
        if (
            $users instanceof Model ||
            $users instanceof Authenticatable
        ) {
            $users = collect([$users]);
        }

        foreach ($users as $user) {
            $user->notify($this->toDatabase());
        }

        return $this;
    }
}
```
Now it's possible to do the following:
```php
// Notify specific user
Notification::sendToDatabase(User::first());

// Notify collected users
Notification::sendToDatabase(User::all());

// Notify users in array
Notification::sendToDatabase([
    User::find(1),
    User::find(2),
]);
```
I imagine treating `null` as "notify every user" would be one step too far?

**Comment by danharrin** [comment]
Yes, I am happy to support arrays and collections of models, also for broadcast notifications. Please make a PR.

> I imagine treating null as "notify every user" would be one step too far?
 
Yes I don't think this is a good idea
